Issues/19 (#53)
trulysimple/tsargp: packages/docs/pages/docs/library/formatter.mdx

messages, including the default group with an empty name. Neither of these methods accept any
parameter.

### Full help

> The `formatFull` method returns a help message that includes more information, as will be explained

Author: ```suggestion
The `formatFull` method returns a help message that includes help sections, as will be explained
```

later in this page. It accepts two optional parameters:

> - `progName` - the name of the program to use in the default usage message
> - `hide` - an object to configure sections that should not be rendered

Author: ```suggestion
- `hide` - an object to configure sections that should not be rendered
- `progName` - the name of the program to use in the default usage message
```

> The `hide` parameter is of type `HideSections`, which contains the following set of optional flags:

Author: ```suggestion
The `hide` parameter, of type `HideSections`, contains the following set of optional flags:
```

## Help format

In addition to the validator instance, the formatter constructor accepts a `HelpConfig` object that
can be used to customize the message format. This configuration will apply to both help sections and

> help entries. It contains a set of optional properties, as described below.

Author: ```suggestion
help entries. It contains a collection of optional properties, as described below.
```

  beginning of the line instead of the end of the option parameter (defaults to `false{:ts}`)
- `usageOptionsAbsolute` whether the indentation level for the usage options should be relative to
  the beginning of the line instead of the end of the program name (defaults to `false{:ts}`)

### Line breaks

The `breaks` property specifies the number of line breaks to insert before help sections or entry
columns. It has the following set of optional properties:

- `intro` - line breaks to insert before the introduction section (defaults to `0{:ts}`)
- `usage` - line breaks to insert before the usage section text (defaults to `2{:ts}`)
- `groups` - line breaks to insert before each option group content (defaults to `2{:ts}`)
- `footer` - line breaks to insert before the footer section (defaults to `2{:ts}`)
- `names` - line breaks to insert before the option names (defaults to `0{:ts}`)
- `param` - line breaks to insert before the option parameter (defaults to `0{:ts}`)
- `descr` - line breaks to insert before the option description (defaults to `0{:ts}`)
- `usageTitle` - line breaks to insert before the usage section heading (defaults to `2{:ts}`)

> - `usageOptions` - line breaks to insert before the usage options (defaults to `0{:ts}`)

Author: ```suggestion
- `usageOptions` - line breaks to insert before the options in the default usage text (defaults to `0{:ts}`)
```
